src/adapters.ts

```diff
@@ -142,3 +143,47 @@ export function redisCacheAdapter(redisCache: RedisLikeCache): Cache {
     },
   };
 }
+
+export interface CloudflareKvCacheConfig {
+  kv: KVNamespace;
+  ctx: ExecutionContext;
+  keyPrefix?: string;
+  name?: string;
+}
+
+export function cloudflareKvCacheAdapter(
+  config: CloudflareKvCacheConfig,
+): Cache {
+  return {
+    name: config.name || 'CloudflareKV',
+    get: async (key) => {
+      const value = await config.kv.get(key, { type: 'text' });
+      if (!value) {
+        return value;
+      }
+      return JSON.parse(value);
+    },
+    set: (key, value) => {
+      const setOperation = async () => {
+        let expirationTtl: number | undefined = totalTtl(value?.metadata);
+        if (expirationTtl === Infinity) {
+          expirationTtl = undefined;
+        } else {
+          expirationTtl = Math.max(Math.ceil(expirationTtl / 1000), 60);
+        }
+
+        await config.kv.put(key, JSON.stringify(value), {
+          expirationTtl: expirationTtl,
+          metadata: value.metadata,
+        });
+      };
+      config.ctx.waitUntil(setOperation());
```

Comment:
That's interesting indeed. Thanks for clarifying!

The thing is that internally, for reporting purposes, cachified is [expecting the `set` action to throw an error when the write fails](https://github.com/kalispera/cachified/blob/main/src/getFreshValue.ts#L61-L74) and similarly a failed `delete` would also throw. 
Therefore I would expect the set and delete methods of this adapter to resolve (with `void | unknown`) once the action was successful or reject on fail.

I understand that this might not be desirable in your case but for the officially provided cache adapters I would expect all of them to behave similarly.